vector/src/main/java/im/vector/riotx/features/ui/UiStateRepository.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.riotx.features.ui
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import im.vector.riotx.features.home.room.list.RoomListFragment
+import javax.inject.Inject
+
+
+/**
+ * This class is used to persist UI state across application restart
+ */
+class UiStateRepository @Inject constructor(private val sharedPreferences: SharedPreferences) {
```

Comment:
Extract interface UIStateRepository and concrete implementation SharedPrefUiStateRepository